### DISCUSSION TITLE
Lookup Flowfields - perform Calcfields upon validating it's parent field

### DISCUSSION THREAD

**Discussion by fvet** [StefanMaron/BusinessCentral.LinterCop#65] [New Rule]
When using 'Lookup' flowfields we always apply a pattern to update the flowfield (Calcfields) when validating it's parent field. This ensures a smooth experiences for the end user when filling the parent field, to automatically see the update of the related (lookup) flowfield.

![image](https://user-images.githubusercontent.com/25268332/137469124-e2fe6daa-26be-4402-8831-c32a40060019.png)

![image](https://user-images.githubusercontent.com/25268332/137469584-b5f63a76-5db3-4700-9d9d-ebe6c2a82c86.png)

It would be nice if we could have a new rule to test:

- On **Table** objects
- On fields with Fieldclass **Flowfield**
- On fields with Calcformula **Lookup**
- On Calcformula including one **field(parent)** reference
- Check the parent fields OnValidate code to contain a Calcfields statement for the current field

**Comment by Arthurvdv** [comment]
While I understand the benefits of this pattern, I am hesitant to include it as a rule in LinterCop.  My primary concern is performance, particularly when inserting or modifying records directly through code without a GUI (page) or when data is created through APIs or other integrations.

A rule for the `OnValidate` trigger on the page itself could make sense if the parent record and one of the 'Lookup' FlowFields are displayed on that page. However, I am uncertain how this would behave when the feature [Add existing table fields to optimize your pages](https://learn.microsoft.com/en-us/dynamics365/release-plan/2023wave2/smb/dynamics365-business-central/add-existing-table-fields-optimize-pages) is used.

I would appreciate any additional feedback or insights on this.
